server/src/main/java/org/opensearch/cluster/node/DiscoveryNode.java

```diff
@@ -470,6 +470,25 @@ public boolean isRemoteStoreNode() {
         return this.getAttributes().keySet().stream().anyMatch(key -> key.startsWith(REMOTE_STORE_NODE_ATTRIBUTE_KEY_PREFIX));
     }
 
+    /**
+     * Returns whether the node is a remote data node.
+     *
+     * @return true if the node contains remote store data attributes, false otherwise
+     */
+    public boolean isRemoteDataNode(){
+        return this.getAttributes().keySet().stream().anyMatch(key -> (key.equals(REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY) ||
+            key.equals(REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY)));
+    }
+
+    /**
+     * Returns whether the node is a remote cluster node.
+     *
+     * @return true if the node contains remote store cluster attributes, false otherwise
+     */
+    public boolean isRemoteClusterNode(){
```

Comment:
```suggestion
    public boolean isClusterStateRemoteBacked(){
```
We can explore other name suggestions too

Reply:
After my suggestions are implemented , we might not be needing this as of now